src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -85,10 +92,21 @@ public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
                 return rsa.SignData(messageBytes, sha);
             }
 #else
-            using (var key = certificate.GetRSAPrivateKey())
+            if (!CertificateToRsaMap.TryGetValue(certificate.Thumbprint, out RSA rsa))
             {
-                return key.SignData(messageBytes, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                if (CertificateToRsaMap.Count == maximumCertificateToRsaMap)
```

Comment:
Could a thread bump the number of certs here to more than 1000 and skip this if? Probably safer if the condition were `>=`

Reply:
yep.